### PR TITLE
Serve from flask

### PR DESCRIPTION
**Pull request by giftofgrub**
This PR allows Flask to serve the build files from React. Intent for this PR is to help in deployment and have the FE and BE running in one port.

Changes I did:
 - ran `npm run build` to build static files
 - serve the build files from Flask
 - configured Axios to have a baseUrl of "/api"
 - all backend routes prepended with "/api"

How to use:
 - Run flask normally with "flask run" in a pipenv.
 - Go to "localhost:5000" to access client side page

One problem that needs reviewing and fixing:
 - When refreshing from client routes other than "/", returns a 404. Right now, the only entry point to the UI is "/".

### CODE REVIEW

**Review by skassam21 — APPROVED** [review]
Looks good - good idea to make the back-end start with /api. 

As for the build folder, I usually gitignore it - since we don't need it in git actually. Just need to add a step to make sure to build it on heroku post pushing - there is a way to add such a step

File: server/app.py [hunk 1]
```diff
@@ -44,15 +44,19 @@ def create_app():
...
-    api.add_resource(ChefResource, '/chefs', '/chefs/<id>')
-    api.add_resource(LoginResource, '/users/login')
-    api.add_resource(MealItemResource, '/meal_items', '/meal_items/<id>')
-    api.add_resource(LogoutResource, '/users/logout')
-    api.add_resource(CuisineResource, '/cuisines')
-    api.add_resource(NotificationResource, '/notifications')
-    api.add_resource(StripeResource, '/create-payment-intent')
-    api.add_resource(OrderResource, '/orders', '/orders/<id>')
+    @app.route('/')
+    def index():
```
Comment (mssiegel) [inline server/app.py]: @app.route('/')
    @app.route('/<path1>')
    @app.route('/<path1>/<path2>')
    def index(**kwargs):
        return app.send_static_file('index.html')
replace with this and you'll be able to reload on any front end route.

Comment (mssiegel) [reply server/app.py]: What this is doing is saying no matter what variables come after the `/` still send index.html
For example, `@app.route('/<path1>/<path2>')` means even ifs `'/anystring/anystring2/'` still serve index.html

File: client/src/api/index.js [hunk 2]
```diff
@@ -1,7 +1,9 @@
 import axios from "axios";
 import Cookies from "js-cookie";
 
-const instance = axios.create();
+const instance = axios.create({
+  baseURL: "/api",
+});
```
Comment (mssiegel) [inline client/src/api/index.js:6]: brilliant job with this

File: client/src/contexts/SocketContext.js [hunk 3]
```diff
@@ -2,7 +2,7 @@ import React from "react";
 
 import io from "socket.io-client";
 
-const socket = io("http://localhost:5000");
+const socket = io("/");
```
Comment (mssiegel) [inline client/src/contexts/SocketContext.js:5]: nice job with this and getting the sockets to work

File: client/build/asset-manifest.json [hunk 4]
```diff
@@ -0,0 +1,21 @@
+{
```
Comment (mssiegel) [inline client/build/asset-manifest.json]: build folder should be ignored from git, in .gitignore switch `/build/` to `*/build/`

File: server/app.py (hunk 1 above)
Comment (giftofgrub) [reply server/app.py]: Thanks Moshe! This was what I was looking for!

**Pull request merged**